### PR TITLE
fix(executor): Deal with the pod watch API call timing out

### PR DESCRIPTION
**Pull request by hermanhobnob**
Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

Tested manually by adding a couple of lines to reduce the api timeout to ~20 seconds and then running a tweaked hello-world workflow with a sleep 60. Ideally this would be an e2e test, but in your youtube contributor video you mentioned needing all the e2e tests to run in under 10 minutes. I'm not sure how much you could compress the times and still have a reliable test for this, so I haven't added an end to end test. Opinions welcome.

### CODE REVIEW

File: workflow/executor/common/wait/wait.go [hunk 1]
```diff
@@ -26,22 +26,32 @@ func UntilTerminated(kubernetesInterface kubernetes.Interface, namespace, podNam
 }
 
 func untilTerminatedAux(podInterface v1.PodInterface, containerID string, listOptions metav1.ListOptions) (bool, error) {
+	for {
+		complete, done, err := doWatch(podInterface, containerID, listOptions)
```
Comment (alexec) [inline workflow/executor/common/wait/wait.go:30]: In English, "complete" means the same thing as "done". Is there a better name to use?

Comment (hermanhobnob) [reply workflow/executor/common/wait/wait.go]: Very fair point. I couldn't think of a better synonym, so I went for something more explicit and swapped the booleans around to match.

**Review by alexec — APPROVED** [review]

**Comment by alexec** [comment]
LGTM

**Comment by hermanhobnob** [comment]
@alexec , is there anything more I need to do with this PR or will you just merge it at some point, when appropriate?

**Comment by alexec** [comment]
@simster7 this needs to be back-ported to v2.12. Would it be best if we label issues that need backporting? @sarabala1979 ?

**Comment by simster7** [comment]
Sure, I am about to make an issue with all the commits since the previous release and was hoping to have you all checkbox which ones you wanted in the release

**Comment by alexec** [comment]
ok, lets do your idea @simster7

**Comment by simster7** [comment]
Let's get this merged in

**Pull request merged**